samples/sensor/tmp116/src/main.c

```diff
@@ -14,13 +14,29 @@ void main(void)
 {
 	const struct device *dev;
 	struct sensor_value temp_value;
+
+	/* offset to be added to the temperature
+	 * only supported by TMP117
+	 */
+	struct sensor_value offset_value;
 	int ret;
 
 	dev = device_get_binding(DT_LABEL(DT_INST(0, ti_tmp116)));
 	__ASSERT(dev != NULL, "Failed to get TMP116 device binding");
 
 	printk("Device %s - %p is ready\n", dev->name, dev);
 
+	/*
+	 * if an offset of 2.5 oC is to be added, set val1 = 2 and val2 = 5
+	 */
```

Comment:
This is wrong. According to `struct sensor_value` documentation you should use `val1 = 2` and `val2 = 500000` in that case. This means that code implementing `SENSOR_ATTR_OFFSET` for tmp117 is also invalid.

Reply:
I changed the sample code and also fixed the driver to use val1 + val2^-6.